### PR TITLE
v6.4: update description for tidb_enable_outer_join_reorder

### PR DESCRIPTION
**Pull request by qiancai**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [ ] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [x] v6.4 (TiDB 6.4 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: system-variables.md [hunk 1]
```diff
@@ -1372,7 +1372,7 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
 
 - 作用域：SESSION | GLOBAL
 - 是否持久化到集群：是
-- 默认值：在 v6.1.0 中为 `ON`，即默认开启。在 v6.1.0 之后的版本中为 `OFF`，即默认关闭。
+- 默认值：在 v6.1.0 中为 `ON`，即默认开启。在 v6.2.0、v6.3.0、v6.4.0 中为 `OFF`，即默认关闭。
 - 自 v6.1.0 起，TiDB 的 [Join Reorder 算法](/join-reorder.md) 开始支持 Outer Join。该变量用于控制这个支持行为。默认关闭，即不启用 Outer Join 的 Join Reorder。
 - 对于从 v6.1.0 之前版本升级到 v6.1.0 及之后的版本，该变量的默认值为 `OFF`。对于从 v6.1.0 版本升级到之后的版本，该变量默认值为 `ON`。
```
Comment (qiancai) [inline system-variables.md:1377]:
```suggestion
- 对于从较低版本升级到当前版本的 TiDB：

    - 如果升级前 TiDB 的版本低于 v6.1.0，升级后该变量的默认值为 `OFF`。
    - 如果升级前 TiDB 的版本等于或大于 v6.1.0，升级后该变量的默认值跟随升级前的设定值。
```

**Review by ran-huang — APPROVED** [review]

**Review by winoros — APPROVED** [review]

**Comment by qiancai** [comment]
Removed the **needs-cherry-pick-release-6.3** label because the v6.3 docs have been archived at https://docs-archive.pingcap.com/zh/tidb/v6.3 and will no longer receive new updates.

**Comment by lilin90** [comment]
Please also help check whether we should keep the descriptions in later versions such as [in TiDB 7.5](https://docs.pingcap.com/zh/tidb/stable/system-variables#tidb_enable_outer_join_reorder-%E4%BB%8E-v610-%E7%89%88%E6%9C%AC%E5%BC%80%E5%A7%8B%E5%BC%95%E5%85%A5).

**Comment by qiancai** [comment]
> Please also help check whether the descriptions in later versions such as [in TiDB 7.5](https://docs.pingcap.com/zh/tidb/stable/system-variables#tidb_enable_outer_join_reorder-%E4%BB%8E-v610-%E7%89%88%E6%9C%AC%E5%BC%80%E5%A7%8B%E5%BC%95%E5%85%A5) are right. image

yes, because this variable was introduced in v6.1.0, upgrading TiDB from a version earlier than v6.1.0 to v6.1.0 will set this variable to its default value of the upgraded TiDB version.

**Comment by qiancai** [comment]
- For v6.2, v6.3, and v6.4, the default value of tidb_enable_outer_join_reorder is OFF. 
- For v6.1, v6.5, and later versions, the default value of tidb_enable_outer_join_reorder is ON. 

As v6.4 docs have been archived, I'll update v6.1 docs instead. 
image

**Comment by qiancai** [comment]
v6.1 PR for doc updates: [15665 ](https://github.com/pingcap/docs-cn/pull/15665/files)

**Pull request closed without merging**